File: src/main/java/com/github/penguin418/todolist/config/TodoUser.java
package com.github.penguin418.todolist.config;

import lombok.AllArgsConstructor;
import lombok.Getter;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.time.LocalDateTime;
import java.util.Collection;

@AllArgsConstructor
public class TodoUser implements UserDetails {
    private final Long userId;
    private final String username;
    private final String password;
    @Getter
    private final String nickname;
    @Getter
    private final LocalDateTime createdDatetime;
    private final Collection<? extends GrantedAuthority> authorities;
    private final Boolean accountNonExpired;
    private final Boolean accountNonLocked;
    private final Boolean credentialsNonExpired;
    private final Boolean enabled;

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return null;
    }

    public Long getUserId() {
        return userId;
    }

    @Override
    public String getPassword() {
        return password;
    }

    @Override
    public String getUsername() {
        return username;
    }

    @Override
    public boolean isAccountNonExpired() {
        return accountNonExpired;
    }

    @Override
    public boolean isAccountNonLocked() {
        return accountNonLocked;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return credentialsNonExpired;
    }

    @Override
    public boolean isEnabled() {
        return enabled;
    }
}

File: src/main/java/com/github/penguin418/todolist/config/TodoUser.kt
package com.github.penguin418.todolist.config

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.UserDetails
import java.time.LocalDateTime

data class TodoUser(
        val userId: Long,
        @JvmField val username: String,
        @JvmField val password: String,
        val nickname: String,
        @JvmField val createdDatetime: LocalDateTime,
        @JvmField val authorities: Collection<GrantedAuthority>,
        @JvmField val accountNonExpired: Boolean,
        @JvmField val accountNonLocked: Boolean,
        @JvmField val credentialsNonExpired: Boolean,
        @JvmField val enabled: Boolean
) : UserDetails {
    override fun getAuthorities(): Collection<GrantedAuthority> {
        return authorities;
    }

    override fun getPassword(): String {
        return password;
    }

    override fun getUsername(): String {
        return username;
    }

    override fun isAccountNonExpired(): Boolean {
        return accountNonExpired;
    }

    override fun isAccountNonLocked(): Boolean {
        return accountNonLocked;
    }

    override fun isCredentialsNonExpired(): Boolean {
        return credentialsNonExpired;
    }

    override fun isEnabled(): Boolean {
        return enabled;
    }
}
